src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task CSharpTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("api")]
+        [InlineData("web")]
+        [InlineData("grpc")]
+        [InlineData("webapi")]
+        [InlineData("webapp")]
+        [InlineData("mvc")]
+        [InlineData("angular")]
+        [InlineData("react")]
+        [InlineData("blazor")]
+        [InlineData("blazorwasm")]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("razorclasslib")]
+        [InlineData("worker")]
+        [InlineData("xunit")]
+        public async Task CSharpTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled), typeof(WindowsOnly))]
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task VisualBasicTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("xunit")]
+        public async Task VisualBasicTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = (templateName, isWindows: ExecutionConditionUtil.IsWindows) switch
+            {
+                (_, isWindows: false) => new string[] { ERR_UndefinedType1 },
+                _ => Array.Empty<string>(),
+            };
```

Comment:
Should this have a comment as to why we're ignoring this in just this case? Because this is a bug, right?

Reply:
I updated the ignored diagnostics to include their message. I agree that there should be none. Skipping templates might be difficult with how tests are discovered.

Reply:
I made it optional.